### PR TITLE
Added a first draft of the keepAlive mechanism for a pod.

### PR DESCRIPTION
**Pull request by meln1k**
# Hello world with keep alive example
## What is this
This is an example of a keep alive mechanism with the existing scheduler.
## Purpose of this PR
I want to grab some attention and discuss the current scheduler interface. Currently it's not that elegant, and to integrate with it we need some workarounds such as queues. 

This PR shows the challenges of integration with the current version.

I think we should discuss a better interface and rewrite it to make it simpler.
## How it works
It's similar to @zen-dog PR, but instead of crashing on pod failure we increase the incarnation number and reschedule pod. Underneath, it has a flow that consumes state updates and transforms them into a stream of pod spec changes.
## What to look at
Have a look on keepAliveWatcher flow, as well as Sink/Source queues used for integrating with the scheduler. 

JIRA issues: DCOS-48963

### CODE REVIEW

File: examples/hello-world/src/main/resources/logback.groovy [hunk 1]
```diff
@@ -1,127 +0,0 @@
-import net.logstash.logback.appender.LoggingEventAsyncDisruptorAppender
```
Comment (alenkacz) [inline examples/hello-world/src/main/resources/logback.groovy]: why?

File: examples/keep-alive-framework/README.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Keep-Alive Example Framework
+
+Run the keep-alove example framework that:
```
Comment (alenkacz) [inline examples/keep-alive-framework/README.md]: alive

File: examples/keep-alive-framework/src/main/scala/com/mesosphere/usi/helloworld/KeepAliveFramework.scala [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+
+  keepAliveStateSource.via(keepAliveWatcher).to(keepAliveSpecSink).run()
+
+
+  val running = Source.single(specsSnapshot)
+    // A trick to make our stream run, we use a workaround to transform the sink of our keep alive mechanism into
+    // a source. If we change the scheduler interface this will be much more elegant
+    .map(snapshot => (snapshot, Source.unfoldResourceAsync[SpecUpdated, SinkQueueWithCancel[SpecUpdated]](
+      () => Future.successful(keepAliveSpecQueue),
```
Comment (jeschkies) [inline examples/keep-alive-framework/src/main/scala/com/mesosphere/usi/helloworld/KeepAliveFramework.scala]: Why do we need this? State updates should just be read, shouldn't they?

**Review by jeschkies — COMMENTED** [review]
First off, many thanks for this example. It is a really great use case. I have to admit that I don't get it though. I don't see where you create a new pod spec. Shouldn't there be a `SpecEvent`? Is it just me or is the interface confusing?

Is there a way we would not leak the Akka stream implementation? I wonder if we could offer both like the Zookeeper client from #51.

**Comment by meln1k** [comment]
I'm closing this PR in favor of #82

**Pull request closed without merging**